libs/math/tests/math/all_tests/activation_functions/softmax.cpp

```diff
@@ -76,6 +76,24 @@ TYPED_TEST(SoftmaxTest, equal_proportion_test)
   }
 }
 
+TYPED_TEST(SoftmaxTest, multi_dimension_test)
```

Comment:
you should test this for both axis == 0 and axis == 1
also you should try a couple of other values (rather than 1/3 everywhere)